### ISSUE TITLE
Perl 5.8.8 fails to compile with gcc 4.2

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8655]
Migrated from [rt.perl.org#40633](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=40633) (status was 'resolved')

Searchable as RT40633$

**Comment by p5pRT** [comment]
### From bero@arklinux.org

#### Created by bero@arklinux.org


This is a bug report for perl from bero@&#8203;arklinux\.org\,
generated with the help of perlbug 1\.35 running under perl v5\.8\.8\.


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
generator output is misinterpreted\.

The configure script runs ok\, but running make results in

make&#8203;: \*\*\* No rule to make target \`\<command\-line>'\, needed by \`miniperlmain\.o'\.  Stop\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=high

Site configuration information for perl v5.8.8:

Configured by Ark Linux at Tue Feb 28 19:39:43 CET 2006.

Summary of my perl5 (revision 5 version 8 subversion 8) configuration:
  Platform:
    osname=linux, osvers=2.6.18-2ark, archname=i386-linux-thread-multi
    uname='linux ch.arklinux.org 2.6.18-2ark #1 wed jan 18 18:40:01 cet 2006 i686 gnulinux '
    config_args='-des -Doptimize=-O2 -march=i586 -mtune=i686 -fomit-frame-pointer -fweb -frename-registers -Dversion=5.8.8 -Dmyhostname=localhost -Dperladmin=root@localhost -Dcc=gcc -Dcf_by=Ark Linux -Dinstallprefix=/usr -Dprefix=/usr -Darchname=i386-linux -Dvendorprefix=/usr -Dsiteprefix=/usr -Duseshrplib -Dusethreads -Duseithreads -Duselargefiles -Dd_dosuid -Dd_semctl_semun -Di_db -Ui_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Duseperlio -Dinstallusrbinperl=n -Ubincompat5005 -Uversiononly -Dpager=/usr/bin/less -isr -Dd_gethostent_r_proto -Ud_endhostent_r_proto -Ud_sethostent_r_proto -Ud_endprotoent_r_proto -Ud_setprotoent_r_proto -Ud_endservent_r_proto -Ud_setservent_r_proto -Dinc_version_list=5.8.7 5.8.6 5.8.5 5.8.4 5.8.3 -Dscriptdir=/usr/bin'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O2 -march=i586 -mtune=i686 -fomit-frame-pointer -fweb -frename-registers',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -I/usr/include/gdbm'
    ccversion='', gccversion='4.1.0 20060222 (prerelease)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lresolv -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lresolv -lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.3.90.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.3.90'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/usr/lib/perl5/5.8.8/i386-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.8:
    /usr/lib/perl5/site_perl/5.8.8/i386-linux-thread-multi
    /usr/lib/perl5/site_perl/5.8.8
    /usr/lib/perl5/site_perl/5.8.7
    /usr/lib/perl5/site_perl/5.8.6
    /usr/lib/perl5/site_perl/5.8.5
    /usr/lib/perl5/site_perl/5.8.4
    /usr/lib/perl5/site_perl/5.8.3
    /usr/lib/perl5/site_perl
    /usr/lib/perl5/vendor_perl/5.8.8/i386-linux-thread-multi
    /usr/lib/perl5/vendor_perl/5.8.8
    /usr/lib/perl5/vendor_perl/5.8.7
    /usr/lib/perl5/vendor_perl/5.8.6
    /usr/lib/perl5/vendor_perl/5.8.5
    /usr/lib/perl5/vendor_perl/5.8.4
    /usr/lib/perl5/vendor_perl/5.8.3
    /usr/lib/perl5/vendor_perl
    /usr/lib/perl5/5.8.8/i386-linux-thread-multi
    /usr/lib/perl5/5.8.8
    .


Environment for perl v5.8.8:
    HOME=/home/bero
    LANG=en_US
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/bin:/bin:/usr/bin:/home/bero/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @doughera88

On Tue\, 31 Oct 2006\, Bernhard Rosenkraenzer wrote&#8203;:

> \# New Ticket Created by  Bernhard Rosenkraenzer 
> \# Please include the string&#8203;:  \[perl \#40633\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=40633 >

> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
> generator output is misinterpreted\.
> 
> The configure script runs ok\, but running make results in
> 
> make&#8203;: \*\*\* No rule to make target \`\<command\-line>'\, needed by \`miniperlmain\.o'\.  Stop\.

Thanks for the report\.  I sometimes wonder how many more different ways 
gcc will invent to say the same thing :\-\)\.

The following patch will fix it \(as will just deleting the offending 
lines\, as you've probably already discovered\.\)  This patch already part of 
the current development release\, but hasn't yet been folded back into the 
maintenance track\.

<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.x/makedepend.SH	2006-06-13 15:31:22.000000000 -0400
+++ perl-5.8.x-andy/makedepend.SH	2006-07-25 14:45:11.000000000 -0400
@@ -167,6 +167,7 @@
             -e '/^#.*<builtin>/d' \
             -e '/^#.*<built-in>/d' \
             -e '/^#.*<command line>/d' \
+            -e '/^#.*<command-line>/d' \
 	    -e '/^#.*"-"/d' \
 	    -e '/: file path prefix .* never used$/d' \
 	    -e 's#\.[0-9][0-9]*\.c#'"$file.c#" \

-- 
```
</details>

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @rgarcia

On 31/10/06\, via RT Bernhard Rosenkraenzer \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
> generator output is misinterpreted\.

Thanks for your report \-\- this has been fixed by change \#28615 to the
development branch of perl :

http&#8203;://public\.activestate\.com/cgi\-bin/perlbrowse?show\_patch=Show\+Patch&patch\_num=28615

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From bero@arklinux.org

#### Created by bero@arklinux.org


This is a bug report for perl from bero@&#8203;arklinux\.org\,
generated with the help of perlbug 1\.35 running under perl v5\.8\.8\.


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
generator output is misinterpreted\.

The configure script runs ok\, but running make results in

make&#8203;: \*\*\* No rule to make target \`\<command\-line>'\, needed by \`miniperlmain\.o'\.  Stop\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=high

Site configuration information for perl v5.8.8:

Configured by Ark Linux at Tue Feb 28 19:39:43 CET 2006.

Summary of my perl5 (revision 5 version 8 subversion 8) configuration:
  Platform:
    osname=linux, osvers=2.6.18-2ark, archname=i386-linux-thread-multi
    uname='linux ch.arklinux.org 2.6.18-2ark #1 wed jan 18 18:40:01 cet 2006 i686 gnulinux '
    config_args='-des -Doptimize=-O2 -march=i586 -mtune=i686 -fomit-frame-pointer -fweb -frename-registers -Dversion=5.8.8 -Dmyhostname=localhost -Dperladmin=root@localhost -Dcc=gcc -Dcf_by=Ark Linux -Dinstallprefix=/usr -Dprefix=/usr -Darchname=i386-linux -Dvendorprefix=/usr -Dsiteprefix=/usr -Duseshrplib -Dusethreads -Duseithreads -Duselargefiles -Dd_dosuid -Dd_semctl_semun -Di_db -Ui_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Duseperlio -Dinstallusrbinperl=n -Ubincompat5005 -Uversiononly -Dpager=/usr/bin/less -isr -Dd_gethostent_r_proto -Ud_endhostent_r_proto -Ud_sethostent_r_proto -Ud_endprotoent_r_proto -Ud_setprotoent_r_proto -Ud_endservent_r_proto -Ud_setservent_r_proto -Dinc_version_list=5.8.7 5.8.6 5.8.5 5.8.4 5.8.3 -Dscriptdir=/usr/bin'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O2 -march=i586 -mtune=i686 -fomit-frame-pointer -fweb -frename-registers',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -Wdeclaration-after-statement -I/usr/local/include -I/usr/include/gdbm'
    ccversion='', gccversion='4.1.0 20060222 (prerelease)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lresolv -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lresolv -lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.3.90.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.3.90'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/usr/lib/perl5/5.8.8/i386-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.8:
    /usr/lib/perl5/site_perl/5.8.8/i386-linux-thread-multi
    /usr/lib/perl5/site_perl/5.8.8
    /usr/lib/perl5/site_perl/5.8.7
    /usr/lib/perl5/site_perl/5.8.6
    /usr/lib/perl5/site_perl/5.8.5
    /usr/lib/perl5/site_perl/5.8.4
    /usr/lib/perl5/site_perl/5.8.3
    /usr/lib/perl5/site_perl
    /usr/lib/perl5/vendor_perl/5.8.8/i386-linux-thread-multi
    /usr/lib/perl5/vendor_perl/5.8.8
    /usr/lib/perl5/vendor_perl/5.8.7
    /usr/lib/perl5/vendor_perl/5.8.6
    /usr/lib/perl5/vendor_perl/5.8.5
    /usr/lib/perl5/vendor_perl/5.8.4
    /usr/lib/perl5/vendor_perl/5.8.3
    /usr/lib/perl5/vendor_perl
    /usr/lib/perl5/5.8.8/i386-linux-thread-multi
    /usr/lib/perl5/5.8.8
    .


Environment for perl v5.8.8:
    HOME=/home/bero
    LANG=en_US
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/bin:/bin:/usr/bin:/home/bero/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @doughera88

On Tue\, 31 Oct 2006\, Bernhard Rosenkraenzer wrote&#8203;:

> \# New Ticket Created by  Bernhard Rosenkraenzer 
> \# Please include the string&#8203;:  \[perl \#40633\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=40633 >

> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
> generator output is misinterpreted\.
> 
> The configure script runs ok\, but running make results in
> 
> make&#8203;: \*\*\* No rule to make target \`\<command\-line>'\, needed by \`miniperlmain\.o'\.  Stop\.

Thanks for the report\.  I sometimes wonder how many more different ways 
gcc will invent to say the same thing :\-\)\.

The following patch will fix it \(as will just deleting the offending 
lines\, as you've probably already discovered\.\)  This patch already part of 
the current development release\, but hasn't yet been folded back into the 
maintenance track\.

<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.x/makedepend.SH	2006-06-13 15:31:22.000000000 -0400
+++ perl-5.8.x-andy/makedepend.SH	2006-07-25 14:45:11.000000000 -0400
@@ -167,6 +167,7 @@
             -e '/^#.*<builtin>/d' \
             -e '/^#.*<built-in>/d' \
             -e '/^#.*<command line>/d' \
+            -e '/^#.*<command-line>/d' \
 	    -e '/^#.*"-"/d' \
 	    -e '/: file path prefix .* never used$/d' \
 	    -e 's#\.[0-9][0-9]*\.c#'"$file.c#" \

-- 
```
</details>

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @rgarcia

On 31/10/06\, via RT Bernhard Rosenkraenzer \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
> generator output is misinterpreted\.

Thanks for your report \-\- this has been fixed by change \#28615 to the
development branch of perl :

http&#8203;://public\.activestate\.com/cgi\-bin/perlbrowse?show\_patch=Show\+Patch&patch\_num=28615

**Comment by p5pRT** [comment]
### From @doughera88

On Tue\, 31 Oct 2006\, Bernhard Rosenkraenzer wrote&#8203;:

> \# New Ticket Created by  Bernhard Rosenkraenzer 
> \# Please include the string&#8203;:  \[perl \#40633\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=40633 >

> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
> generator output is misinterpreted\.
> 
> The configure script runs ok\, but running make results in
> 
> make&#8203;: \*\*\* No rule to make target \`\<command\-line>'\, needed by \`miniperlmain\.o'\.  Stop\.

Thanks for the report\.  I sometimes wonder how many more different ways 
gcc will invent to say the same thing :\-\)\.

The following patch will fix it \(as will just deleting the offending 
lines\, as you've probably already discovered\.\)  This patch already part of 
the current development release\, but hasn't yet been folded back into the 
maintenance track\.

<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.x/makedepend.SH	2006-06-13 15:31:22.000000000 -0400
+++ perl-5.8.x-andy/makedepend.SH	2006-07-25 14:45:11.000000000 -0400
@@ -167,6 +167,7 @@
             -e '/^#.*<builtin>/d' \
             -e '/^#.*<built-in>/d' \
             -e '/^#.*<command line>/d' \
+            -e '/^#.*<command-line>/d' \
 	    -e '/^#.*"-"/d' \
 	    -e '/: file path prefix .* never used$/d' \
 	    -e 's#\.[0-9][0-9]*\.c#'"$file.c#" \

-- 
```
</details>

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @rgarcia

On 31/10/06\, via RT Bernhard Rosenkraenzer \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> Perl 5\.8\.8 fails to compile with gcc 4\.2 \-\- apparently the dependency
> generator output is misinterpreted\.

Thanks for your report \-\- this has been fixed by change \#28615 to the
development branch of perl :

http&#8203;://public\.activestate\.com/cgi\-bin/perlbrowse?show\_patch=Show\+Patch&patch\_num=28615
